src/main/resources/assets/js/app/issue/IssueType.ts

```diff
@@ -13,4 +13,15 @@ export class IssueTypeFormatter {
             return null;
         }
     }
+
```

Comment:
This was probably done in accordance with `IssueStatus` class (which is also wrong), but...

Both `parseType` and `parseTypeName` are now very fragile - both will break if we add a new issue type. Also, we should always return `STANDARD` as default, to make sure UI won't break if for example someone creates a new issue with unknown issue type via API.

`parseType` could simply be `return IssueType[value] || IssueType.STANDARD;`
`parseTypeName`: `return (IssueType[value] || IssueType[IssueType.STANDARD]).toLowerCase();` (you'll have to change "-" to "_" in expected return value but this shouldn't be a problem

Reply:
Yes. I've refactored `IssueTypeFormatter` and `IssueStatusFormatter`, which made some functions simpler. However, `parseType` may return `undefined` instead of the default value. I think it's more obvious if this function will be used elsewhere and some invalid value is passed. The default value is set in the `IssueSummaryBuilder` if the server returns unknown value or `null` instead of type.